### ISSUE TITLE
UDF vs UDP confusion

### ISSUE THREAD

**Issue by soxofaan** [Open-EO/openeo-api#353]
When supporting new openEO users I noticed that it is very easy to confuse the concepts of UDFs (user defined functions) and UDPs (user defined processes). Not only their abbreviation is very close, also the conceptual difference between "function" and "process" in the context of openEO is hard to grasp when you are just starting to learn openEO (e.g. we probably explain somewhere that an openEO "process" is like a "function" in regular programming language).

I think we should consider renaming at least one of them.
I don't have a proposal at the moment, but just wanted to raise this issue here to start a discussion.

**Comment by m-mohr** [comment]
Indeed, that can be confusing. I'm trying to avoid UDP and just write "process" in commmunication, but on the other hand I never liked the term UDF. That seems to be taken from database terminology, but "function" seems to be wrong with regards to what we offer...

**Comment by edzer** [comment]
Maybe, instead of UDF, "script", as of "Python- or R-script"?

**Comment by m-mohr** [comment]
@edzer I thought the same, but would that just be "openEO script" or "UDS" then?

**Comment by soxofaan** [comment]
to estimate scope and impact:

- openeo-api/openapi.yaml: 
    - 76 occurrences of `user.defined.*?process`, no occurrence of `UDP`
    - 22 occurrences of `UDF`, 1 occurrence of `user.defined.*?function`
- openeo-processes:
    - 4 occurences of  `user.defined.*?process`, no occurrence of `UDP`
    - 63 occurences of `UDF`, 1 occurrence of `user.defined.*?function`
- openeo-python-client (user facing code and docs):
    -   60 occurrences of `user.?defined.*?process`, no occurence of `UDP`
    - 206 occurences of `UDF`, 6 occurences of  `user.?defined.*?function`

**Comment by m-mohr** [comment]
This issue should get through PSC vote if there's a desire to change one of the terms, I think.

**Comment by m-mohr** [comment]
Just realized: Renaming UDF would "require" a rename for the endpoint `/udf_runtimes`.
